dsnp-wallet/dsnp-wallet/Source/ViewControllers/ProfileViewController.swift

```diff
@@ -5,10 +5,366 @@
+        
+        let profileHeaderView = ProfileHeaderView(parent: self)
+        profileHeaderView.delegate = self
+        self.profileHeaderView = profileHeaderView
+    
+        let profileUsernameView = getTextfieldViewWith(titleLabel: profileText, placeholder: "Add Profile Name")
+        let bioView = getTextfieldViewWith(titleLabel: bioText, placeholder: "Add Bio")
+        let permissionsView = getPermissionsView()
+        
+        stackView.addArrangedSubview(profileHeaderView)
+        stackView.addArrangedSubview(SharedSpacer(height: 6))
+        stackView.addArrangedSubview(profileUsernameView)
+        stackView.addArrangedSubview(bioView)
+        stackView.addArrangedSubview(SharedSpacer(height: 3))
+        stackView.addArrangedSubview(permissionsView)
+        
+        setSaveBtn()
+        setKeyboard()
+    }
+    
+    private func getScrollableStackView() -> UIStackView {
+        let scrollView = UIScrollView()
+        view.addSubview(scrollView)
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+        scrollView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        scrollView.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor).isActive = true
+        scrollViewBottomConstraint = NSLayoutConstraint(item: scrollView,
+                                                      attribute: .bottom,
+                                                      relatedBy: .equal,
+                                                      toItem: self.view.layoutMarginsGuide,
+                                                      attribute: .bottom,
+                                                      multiplier: 1,
+                                                      constant: 0)
+        scrollViewBottomConstraint.isActive = true
+        
+        let bottomInset = (self.navigationController?.navigationBar.frame.height ?? 0.0) + saveBtn.frame.height + 60
+        scrollView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: bottomInset, right: 0)
+        
+        let embeddedView = UIView()
+        scrollView.addSubview(embeddedView)
+        embeddedView.layoutAttachAll(to: scrollView.contentLayoutGuide)
+        embeddedView.widthAnchor.constraint(equalTo: scrollView.frameLayoutGuide.widthAnchor).isActive = true
+
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.distribution = .fill
+
+        embeddedView.addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.leadingAnchor.constraint(equalTo: embeddedView.leadingAnchor).isActive = true
+        stackView.centerXAnchor.constraint(equalTo: embeddedView.centerXAnchor).isActive = true
+        stackView.topAnchor.constraint(equalTo: embeddedView.layoutMarginsGuide.topAnchor, constant: 10).isActive = true
```

Comment:
We're using an offset of 10 here. Should we just add this to the bottomInset value? Or are they serving different purposes?

Reply:
Good catch, this wasn't necessary anymore so I removed it.